src/components/molecules/LocationMap.tsx

```diff
@@ -1,32 +1,72 @@
 import 'leaflet-css'
 import React, { FunctionComponent } from 'react'
 import L from 'leaflet'
-import { Map, Marker, TileLayer } from 'react-leaflet'
+import { Map, TileLayer } from 'react-leaflet'
+import AnywayMarker from '../atoms/AnywayMarker'
 
 // TODO: Move to init.service once it's merged.
 L.Icon.Default.mergeOptions({
     iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
     iconUrl: require('leaflet/dist/images/marker-icon.png'),
     shadowUrl: require('leaflet/dist/images/marker-shadow.png')
 })
+const lIconSizes = { iconSize: [25, 41], iconAnchor: [12, 41], popupAnchor: [1, -34], shadowSize: [41, 41] }
+const iconSize = lIconSizes;
+
 
 interface IProps {
+    data: any[],
     marker: { lat: number; lng: number }
 }
 
 const INITIAL_ZOOM = 13
 const WRAPPER_STYLES = { height: '100%', width: '100%' }
 
-const LocationMap: FunctionComponent<IProps> = ({ marker }) => {
+const LocationMap: FunctionComponent<IProps> = ({ data, marker }) => {
+    let markers = data.map((x: any) => {
+        if (x.latitude !== null && x.longitude !== null) {
+            return <AnywayMarker markerdata={x} />
+        }
+        else return null;
+    });
+    const bounds = setBounds(data)
+
     return (
-        <Map center={marker} zoom={INITIAL_ZOOM} style={WRAPPER_STYLES}>
+        <Map center={marker} bounds={bounds} zoom={INITIAL_ZOOM} style={WRAPPER_STYLES}>
             <TileLayer
                 url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
                 attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
             />
-            <Marker position={marker} />
+            {markers}
         </Map>
     )
 }
-
+const setBounds = (data: any[]) => {
+    var bounds;
+    var corner1 = L.latLng(29.50, 34.22)
+    var corner2 = L.latLng(33.271, 35.946)
+    var arr:L.LatLng[] = [];
+    var lastPoint:L.LatLng = L.latLng(0,0) ;
+    data.forEach(x => {
+        if (x.latitude !== null && x.longitude !== null) {
+            let p = new L.LatLng(x.latitude, x.longitude);
+            if ((lastPoint.lat ==0 && lastPoint.lng ==0 ) || x.latitude != lastPoint.lat || x.longitude != lastPoint.lng)
+            {
+                arr.push(p)
+                //prevent insertion of dupliacte same point
+                lastPoint =  p;
+            }
+        }
+    });
+    //bounds for single point
+    if (arr.length == 1){
+        arr=[];
+        arr.push( L.latLng(lastPoint.lat+0.01, lastPoint.lng +0.01))
+        arr.push( L.latLng(lastPoint.lat-0.01, lastPoint.lng -0.01))
+    }
+    if (arr.length < 2)
```

Comment:
The logic in the block above will never be returned, because this will run and replace it. Did you mean `arr.length < 1`?

Reply:
if we have o or zero points - bounds will return error, we need 2 points . 
in case of zero points this coed will run - for example accidents without lat-lon info .

Reply:
Still, if the array length is one, then you:
- Run the previous block, which resets the array and pushes two latLng objects into it
- Replace the array you've just created with `[corner1, corner2]`

Reply:
no, we have 3 cases 
- get 0 points - return default points of all israel,
- get 1 point  - replace it with 2 poiints around the origainl point
- get 2 or more points -  calculate bounds for all.
i have tested all 3 cases with the data we get from the server